### PR TITLE
Ap sm delete item

### LINKED ISSUE #12: 12. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future
## Summary
Users might make a mistake when entering an item, or may decide they won’t be buying a certain item again in the future. Allowing them to delete list items will help them keep their lists tidy.
 
## Acceptance criteria
- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Notes:
* Activating the “Delete” button should show a confirmation dialog. We want to give users a chance to confirm “destructive” actions.
* It might be tempting to create your own modal dialog component to handle confirmation, but modals are fraught with accessibility concerns, so consider using a simple [JavaScript confirm dialog](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm).

### PR DESCRIPTION
**Pull request by andiedoescode**
## Description

- Added delete button next to each ListItem.
- `deleteHandler` function for the buttom to trigger user confirmation window, then the `deleteItem` function.
- Completed `deleteItem` function in `firebase.js`

*Note: While completing this issue, lead to potential creation of new tickets (deleting lists, user feedback/error handling UI)*

## Related Issue

Closes #12 

## Acceptance Criteria

- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Type of Changes

`new feature`

### Before

image

### After

image
image
image

## Testing Steps / QA Criteria
- Open deployment preview link OR navigate to branch `ap-sm-delete-item`.
- Log in to account.
- Select 'dinner' list.
- Navigate to List page.
- Select an item to delete - click the 'X' next to the checkbox.
- Confirm that dialog box pops up to request confirmation of item deletion with the selected item's name.
- When you click 'OK', the item should disappear from the List page.
- Select another item to delete. When the confirmation dialog pops up, click 'Cancel'.
- Ensure that the item does not disappear from the list.

### CODE REVIEW

**Review by DevinaG007 — DISMISSED** [review]
Looks good! I do have one nit: I think the delete button could be made more accessible since we're using an X for the text of the button, and it might not be 100% clear. Check out this [article on accessible buttons](https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/) that I found. Otherwise, lovely work as always!

Dancing bear

File: src/api/firebase.js [hunk 1]
```diff
@@ -239,10 +240,12 @@ export async function updateItem(listPath, itemID, isChecked) {
...
 }
 
-export async function deleteItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to delete an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function deleteItem(listPath, itemID) {
+	// Note - firebase deleteDoc will remove the entire collection if the item being removed is the last in said collection!
```
Comment (etiry) [inline src/api/firebase.js:244]: question: Just curious if that has any implications for any other functionality? I tried adding one item to one of my own lists, deleting it, and re-adding it and it didn't seem to break anything 🤞🏻

File: src/components/ListItem.jsx [hunk 2]
```diff
@@ -29,6 +29,13 @@ export function ListItem({ listPath, item }) {
 		purchaseItem();
 	};
 
+	const deleteHandler = async (e) => {
+		// Note: Should we add more user feedback when items are successfully deleted? Some might further interrupt usability.
+		if (window.confirm(`Are you sure you'd like to delete ${item.name}?`)) {
```
Comment (etiry) [inline src/components/ListItem.jsx:34]: praise: I like how you display the item name in the confirmation dialog, makes it easier to confirm you're deleting the right thing!

**Review by etiry — DISMISSED** [review]
Good work--simple and straightforward solution!

File: src/api/firebase.js (hunk 1 above)
Comment (andiedoescode) [reply src/api/firebase.js:244]: From some further digging, we noticed the pattern from the dinner list, and others that get created. Our lists are stored as documents within each of our user id collections. These documents then contain the 'items' items, which then has the documents for each item in the list. Which theoretically means it only removes the 'items' collection, but doesn't touch the list doc itself. AFAIK, there shouldn't be any functionality implications for now.

**Review by raaynaldo — DISMISSED** [review]
LGTM 🎉

File: src/api/firebase.js (hunk 1 above)
Comment (sdmisra) [reply src/api/firebase.js:244]: > there shouldn't be any functionality implications for now

I think it's mostly an aspect that will be useful for us to know as the developers of this project, as we look at documentation / seek to build on this structure that we have!

**Comment by andiedoescode** [comment]
> Looks good! I do have one nit: I think the delete button could be made more accessible since we're using an X for the text of the button, and it might not be 100% clear. Check out this [article on accessible buttons](https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/) that I found. Otherwise, lovely work as always!
> 
> Dancing bear [ Dancing bear ](https://camo.githubusercontent.com/fb75fba5c71627dae89b5d1f206998dc21201c6033b66137e0985bdbd5a2c39d/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f76312e59326c6b505463354d4749334e6a45784f574e726554426d5a7a6332613356786332313061475131656e527363546c3463544e326444523362574a324d5468364f58453262435a6c634431324d563970626e526c636d35686246396e61575a66596e6c666157516d593351395a772f74735833594d57597a44506a4141526665672f67697068792e676966) [ ](https://camo.githubusercontent.com/fb75fba5c71627dae89b5d1f206998dc21201c6033b66137e0985bdbd5a2c39d/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f76312e59326c6b505463354d4749334e6a45784f574e726554426d5a7a6332613356786332313061475131656e527363546c3463544e326444523362574a324d5468364f58453262435a6c634431324d563970626e526c636d35686246396e61575a66596e6c666157516d593351395a772f74735833594d57597a44506a4141526665672f67697068792e676966)

Very valid nit and likely good practice to incorporate the accessibility assessment throughout the whole process!

**Review by raaynaldo — APPROVED** [review]

**Review by DevinaG007 — APPROVED** [review]
Looks good! 
fabulous gif tony stark

**Pull request merged**